packages/utils/src/normalize.ts

```diff
@@ -51,170 +53,173 @@ export function normalizeToSize<T>(
+  // pass-through for everything else.)
+  const source = convertToPlainObject(value);
+
+  // Create an accumulator that will act as a parent for all future iterations of this branch, and keep track of the
+  // number of properties/entries we add to it
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const acc: { [key: string]: any } = Array.isArray(value) ? [] : {};
+  let numAdded = 0;
 
-  // If we already walked that branch, bail out, as it's circular reference
+  // If we've already visited this branch, bail out, as it's circular reference
   if (memoize(value)) {
     return '[Circular ~]';
   }
 
-  let propertyCount = 0;
-  // Walk all keys of the source
+  // visit all keys of the source
   for (const innerKey in source) {
     // Avoid iterating over fields in the prototype if they've somehow been exposed to enumeration.
     if (!Object.prototype.hasOwnProperty.call(source, innerKey)) {
       continue;
     }
 
-    if (propertyCount >= maxProperties) {
+    if (numAdded >= maxProperties) {
       acc[innerKey] = '[MaxProperties ~]';
       break;
     }
 
-    propertyCount += 1;
+    // Recursively visit all the child nodes
+    const innerValue: unknown = source[innerKey];
+    acc[innerKey] = visit(innerKey, innerValue, depth - 1, maxProperties, memo);
 
-    // Recursively walk through all the child nodes
-    const innerValue: UnknownMaybeWithToJson = source[innerKey];
-    acc[innerKey] = walk(innerKey, innerValue, depth - 1, maxProperties, memo);
+    numAdded += 1;
   }
 
-  // Once walked through all the branches, remove the parent from memo storage
+  // Once we've visited all the branches, remove the parent from memo storage
   unmemoize(value);
 
   // Return accumulated values
   return acc;
 }
 
-/**
- * Transform any non-primitive, BigInt, or Symbol-type value into a string. Acts as a no-op on strings, numbers,
- * booleans, null, and undefined.
- *
- * @param value The value to stringify
- * @returns For non-primitive, BigInt, and Symbol-type values, a string denoting the value's type, type and value, or
- *  type and `description` property, respectively. For non-BigInt, non-Symbol primitives, returns the original value,
- *  unchanged.
- */
-function serializeValue(value: any): any {
-  // Node.js REPL notation
-  if (typeof value === 'string') {
-    return value;
-  }
-
-  const type = Object.prototype.toString.call(value);
-  if (type === '[object Object]') {
-    return '[Object]';
-  }
-  if (type === '[object Array]') {
-    return '[Array]';
-  }
-
-  // `makeSerializable` provides a string representation of certain non-serializable values. For all others, it's a
-  // pass-through.
-  const serializable = makeSerializable(value);
-  return isPrimitive(serializable) ? serializable : type;
-}
+// TODO remove this in v7 (we don't use it anywhere, but it's a public method)
+export { visit as walk };
```

Comment:
Can we add to the comment to also remove `export` from the `visit` method with v7? Just so we don't forget.

Reply:
I actually wan't planning to un-public it (not a real word, but you know what I mean) - just let it live only under its new name. That said, I guess there's no reason we need to be exporting it. Given the export here, I can just remove it from `visit` itself now.